HealthcareSystem/PatientWebApp/Controllers/ExaminationController.cs

```diff
@@ -65,5 +68,29 @@ public ActionResult AdvanceSearchExaminations(ExaminationSearchDTO examinationSe
 
         }
 
+        /// <summary>
+        /// / updating feedbacks status (property: IsPublished) to published
+        /// </summary>
+        /// <param name="id">id of the object to be changed</param>
+        /// <returns>if alright returns code 200(Ok), if not 400(bed request), if connection lost returns 500</returns>
+        [HttpPut("cancel/{id}")]
+        public ActionResult CancelExamination(int id)
+        {
+            try
+            {
+                _examinationValidator.CheckIfExaminationCanBeCanceled(id);   
+            }
+            catch (ValidationException exception)
+            {
+                return BadRequest(exception.Message);
+            }
+            catch (DatabaseException exception)
+            {
+                return StatusCode(500, exception.Message);
+            }
+            _examinationService.CancelExamination(id);
+            return Ok();
+        }
```

Comment:
So, one quick thing to fix - the Service method call can also throw exceptions. When writing Controllers, I would generally put the entire happy path in the try catch, so this would be something like:
```
            try
            {
                _examinationValidator.CheckIfExaminationCanBeCanceled(id);   
                _examinationService.CancelExamination(id);
                return Ok();
            }
            catch (ValidationException exception)
            {
                return BadRequest(exception.Message);
            }
            catch (DatabaseException exception)
            {
                return StatusCode(500, exception.Message);
            }
```
Another thing that kind of bothers me is that you fetch the Examination from the database twice, once from the Validator, and once from the Service. Basically, here's how I would fix that - get rid of the Validator in this case entirely. Instead, make a Cancel method in Examination. Have that method check whether the state transition can happen or not, and throw an exception if it can't. Actually, I feel like that's approximately what we'll be asked to do with pretty much all validations for DDD anyway - so because of that I also don't mind if you keep it as is for now and refactor later.

Reply:
Fixed the Controller method in 8d297a7a201cb557715f22f218845f03c8793464. Moving the Validation postponed until next sprint, since we'll start applying DDD then.